lighthouse-core/fraggle-rock/config/filters.js

```diff
@@ -20,6 +20,43 @@ const baseArtifactKeySource = {
 
 const baseArtifactKeys = Object.keys(baseArtifactKeySource);
 
+/**
+ * Returns the set of audit IDs used in the list of categories.
+ * If `onlyCategories` is not set, this function returns the list of all audit IDs across all
+ * categories.
+ *
+ * @param {LH.Config.FRConfig['categories']} allCategories
+ * @param {string[] | undefined} onlyCategories
+ * @return {string[]}
+ */
+function getAuditIdsInCategories(allCategories, onlyCategories) {
+  if (!allCategories) return [];
+
+  onlyCategories = onlyCategories || Object.keys(allCategories);
+  const categories = onlyCategories.map(categoryId => allCategories[categoryId]);
+  const auditRefs = categories.flatMap(category => category.auditRefs);
+  return auditRefs.map(auditRef => auditRef.id);
```

Comment:
Not sure if it makes a difference down the line, but can we dedupe audit ids for cases like `viewport` which is in SEO and PWA.

Reply:
sure, it's already put into a set before use below, but makes sense for this api too :)